core/src/autogluon/core/scheduler/mo_hyperband.py

```diff
@@ -0,0 +1,170 @@
+import logging
+import threading
+import numpy as np
+import multiprocessing as mp
+
+from .hyperband import HyperbandScheduler
+from .reporter import MODistStatusReporter
+from ..utils.default_arguments import filter_by_key
+
+__all__ = ['MOHyperbandScheduler']
+
+logger = logging.getLogger(__name__)
+
+_ARGUMENT_KEYS = {
+    'objectives', 'scalarization_options'
+}
+
+
+class MOHyperbandScheduler(HyperbandScheduler):
+    r"""Implements different variants of asynchronous multi-objective Hyperband
+    by extending the standard HyperbandScheduler using random scalarizations. 
+    The implemented method is described in detail in:
+
+        Schmucker, Donini, Perrone, Zafar, Archambeau (2020)
+        Multi-Objective Multi-Fidelity Hyperparameter Optimization with 
+        Application to Fairness
+        https://meta-learn.github.io/2020/papers/24_paper.pdf
+
+    This class supports the full range of parameters of the standard
+    HyperbandScheduler. The 'objectives' replaces the 'reward_attr' parameter.
+    Here we only explain the multi-objective specific parameters. For the full 
+    list of available options refer to HyperbandScheduler.
+
+    Parameters
+    ----------
+    objectives : dict
+        Dictionary with the names of objectives of interest. The corresponding
+        values are allowed to be either "MAX" or "MIN" and indicate if an 
+        objective is to be maximized or minimized.
+    scalarization_options : dict
+        Contains arguments for the underlying scalarization algorithm. 
+        Available algorithms are "random_weights" and "parego".
+
+    Examples
+    --------
+    examples/mo_hpo/mo_hyperband.ipynb
+    """
+
+    def __init__(self, train_fn, **kwargs):
+        self._scalarization_options = kwargs["scalarization_options"]
+        self._objectives = kwargs["objectives"]
+        self._sign_vector = _prepare_sign_vector(self._objectives)
+        kwargs["reward_attr"] = "_SCALARIZATION"
```

Comment:
is it possible that `kwargs["reward_attr"]` overwrites silently?

Reply:
The only case in which a silent overwrite can happen is when the user explicitly specifies "_SCALARIZATION" as an objective in the dictionary. We can use an assert statement to prevent the user from doing this and provide feedback.
```suggestion
        assert "_SCALARIZATION" not in kwargs["objectives"], "_SCALARIZATION" \
            "is a protected value. Please refrain from using it."
        kwargs["reward_attr"] = "_SCALARIZATION"
```